### PR TITLE
Child User Authentication

### PR DESCRIPTION
**Pull request by shadinaif**
This is to add `edraak_users` app. It is a copy of https://github.com/Edraak/edx-platform/pull/349 with some refactoring

### CODE REVIEW

**Comment by shadinaif** [comment]
We'll squash it after your review @OmarEdraak

File: common/djangoapps/edraak_dummy/urls.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import url
+from edraak_dummy.views import SetCSRFDummyView
+
+urlpatterns = [
+    url(
+        '^edraak_set/csrf',
+        SetCSRFDummyView.as_view(),
```
Comment (ghost) [inline common/djangoapps/edraak_dummy/urls.py]: Thanks @shadinaif, Could you please remove the `edraak_dummy` app? It was there to solve a bug in Dogwood that Ali and I managed to resovle in Hawthorn.

Sorry for making you to go into the trouble of migrating it.

File: common/djangoapps/student/constants.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CHILD_USER_PERMISSION_GROUP = u'CHILD_USER'
+CHILD_OF_USER_PERMISSION_GROUP_PREFIX = u'CHILD_OF_{}'
+LIMITED_USER_GROUPS = [CHILD_USER_PERMISSION_GROUP,]
```
Comment (ghost) [inline common/djangoapps/student/constants.py]:
```suggestion
LIMITED_USER_GROUPS = [CHILD_USER_PERMISSION_GROUP]
```

File: lms/envs/aws.py [hunk 3]
```diff
@@ -690,6 +690,8 @@
 
 INSTALLED_APPS += ('edraak_specializations',)
 
+INSTALLED_APPS += ('edraak_dummy',)
```
Comment (ghost) [inline lms/envs/aws.py]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]
a couple of changes

File: common/djangoapps/student/constants.py (hunk 2 above)
Comment (shadinaif) [reply common/djangoapps/student/constants.py]: Done

File: lms/envs/aws.py (hunk 3 above)
Comment (shadinaif) [reply lms/envs/aws.py]: Done

File: common/djangoapps/edraak_dummy/urls.py (hunk 1 above)
Comment (shadinaif) [reply common/djangoapps/edraak_dummy/urls.py]: Done

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Thanks @shadinaif! I've merged this PR.

**Pull request merged**